index.js

```diff
@@ -21,12 +21,19 @@ module.exports = function generateRequireForUserCode (scopedDirs, options) {
   addPaths(baseModule)
 
   _.forEach(require.extensions, function (extensionLoader, extension) {
-    const original = extensionLoader
-
-    require.extensions[extension] = function requireThatAddsUserCodeDirs (m, filename) {
-      addPaths(m)
-
-      return original(m, filename)
+    /*
+    Node.JS has 3 extensions: '.js', 'json' and 'node'. Each one has a loader which is a function.
+    There are tools, like jest, that add extensions (like .mjs) without a proper loader.
+    This `if` handles those cases, mostly for test environments.
+    */
```

Comment:
```suggestion
    /*
    Node.JS has 3 extensions: '.js', 'json' and 'node'. Each one has a loader which is a function.
    We add our own extension '.jsw', with a loader function.
    There are tools, like jest, that add extensions (like .mjs) without a proper loader.
    This `if` handles those cases, mostly for test environments.
    */
```

Feel free to change the phrasing